oci/casext/gc.go

```diff
@@ -35,7 +38,7 @@ import (
 // functions. In other words, it assumes it is the only user of the image that
 // is making modifications. Things will not go well if this assumption is
 // challenged.
-func (e Engine) GC(ctx context.Context) error {
+func (e Engine) GC(ctx context.Context, policies ...GCPolicy) error {
```

Comment:
I'm wondering whether it'd be better to have this just take a single function -- callers can always write their own logic for multiple policies but we don't know whether a caller would prefer that we require *all* policies to match or just one policy to match. In addition, what if the caller wants all of the policy functions to be executed, even if one said to keep the blob?

But if we do that then we should create a new `GC` function called `GCWithPolicy` (or something) which takes a single function as an argument, and leave the `GC` signature the same.

Reply:
Yeah, that all sounds reasonable; GCWithPolicy() would save a lot of code updates everywhere for everyone :)

Reply:
Though in fairness (and I missed this when I first looked) -- the `GC(ctx, ...)` approach actually doesn't break anyone's builds because not passing any extra arguments gives you a `nil` slice.

Reply:
Tried to make sure there is no behaviour or API changes as much as possible.